File: 2020/07/14-MoveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int arrSize = nums.length;
        
        int k = 0;
        
        for(int i = 0; i < arrSize; ++i) {
            if (nums[i] != 0)
                nums[k++] = nums[i];
        }
        
        for (int j = k; j < arrSize; ++j) {
            nums[j] = 0;
        }
    }
}

File: 2020/07/14-MoveZeroes.kt
class Solution {
    fun movezeroes(nums: IntArray): Unit {
        var k = 0

        for (x in 0..(nums.size-1)) {
            if (nums[x] != 0) {
                nums[k++] = nums[x]
            }
        }

        for (y in k..(nums.size-1)) {
            nums[y] = 0
        }
    }
}